include/common/bitdepth.rs

```diff
@@ -296,11 +292,18 @@ macro_rules! bd_fn {
             }
         }
     }};
+
+    ($BD:ty, $name:ident, $asm:ident) => {{
+        use paste::paste;
+
+        paste! {
+            match BD::BPC {
+                BPC::BPC8 => [<dav1d_ $name _8bpc_ $asm>],
+                BPC::BPC16 => [<dav1d_ $name _16bpc_ $asm>],
+            }
+        }
+    }};
```

Comment:
Actually, I think if you don't add this version and declare the asm `fn`s inline (like I did for the original one), you won't get the `#[allow(dead_code)]` problem and we don't have to duplicate the `#[cfg]` logic.  Could that work?

Reply:
I don't think it's worth reworking this just to avoid the `#[allow(dead_code)]` on the function decls. What we have now works and matches how the original C was declaring the functions.

I'll go ahead and add the `fn_identity` macro in `bitdepth.rs`, though.

Reply:
Wouldn't it be simpler to only have that logic once (in the `dsp_init` `fn`s), rather that in the `dsp_init` `fn`s, where it's exact, and the pure declarations, where it's overly broad?  It'd be both simpler, still follow the C logic, and wouldn't declare and bunch of non-existent `fn`s.

Reply:
You may be right that it'd ultimately be cleaner, but I don't think it's worth the time it'll take to rework this. I'm going to merge this as-is to keep things moving.